app/assets/javascripts/sumofus/backbone/fundraiser_bar.js

```diff
@@ -0,0 +1,256 @@
+      let digits = (this.donationAmount === Math.floor(this.donationAmount)) ? 0 : 2;
+      let donationAmount = `${currencySymbol}${this.donationAmount.toFixed(digits)}`;
+      let buttonText = `<span class="fa fa-lock"></span><span>Donate ${donationAmount}</span>`
+      this.$('.fundraiser-bar__display-amount').text(donationAmount);
+      this.$('.fundraiser-bar__submit-button').html(buttonText);
+    } else {
+      this.changeStep(1);
+    }
+  },
+
+  triggerStepChange (e) {
+    const targetStep = this.$(e.target).parent().data('step');
+    if (targetStep < this.currentStep) {
+      this.changeStep(targetStep);
+    }
+  },
+
+  changeStep (targetStep) {
+    this.changeStepPanel(targetStep);
+    this.changeStepNumber(targetStep);
+    this.currentStep = targetStep;
+  },
+
+  changeStepPanel (targetStep) {
+    this.$('.fundraiser-bar__step-panel').addClass('hidden-closed');
+    this.$(`.fundraiser-bar__step-panel[data-step="${targetStep}"]`).removeClass('hidden-closed');
+  },
+
+  changeStepNumber (targetStep) {
+    $.each(['number', 'name'], (ii, part) => {
+      this.$(`.fundraiser-bar__step-${part}`).
+        removeClass(`fundraiser-bar__step-${part}--past`).
+        removeClass(`fundraiser-bar__step-${part}--current`).
+        removeClass(`fundraiser-bar__step-${part}--upcoming`);
+      this.$(`.fundraiser-bar__step-${part}`).each((ii, el) => {
+        const step = this.$(el).parent().data('step');
+        if ( step < targetStep ) {
+          $(el).addClass(`fundraiser-bar__step-${part}--past`);
+        } else if ( step == targetStep) {
+          $(el).addClass(`fundraiser-bar__step-${part}--current`);
+        } else {
+          $(el).addClass(`fundraiser-bar__step-${part}--upcoming`);
+        }
+      });
+    });
+  },
+
+  // for testing without waiting on braintree API
+  fakeNonceSuccess (fakeData) {
+    this.paymentMethodReceived()(fakeData);
+  },
+
+  paymentMethodReceived () {
+    return (data) => {
+      this.nonce = data.nonce;
+      this.submitDonation();
+    }
+  },
+
+  submitDonation () {
+    $.post(`/api/braintree/pages/${this.pageId}/transaction`, {
+      payment_method_nonce: this.nonce,
+      amount:               this.donationAmount,
+      user:                 this.serializeUserForm(),
+      currency:             this.currency,
+      recurring:            this.readRecurring()
+    }).done(this.transactionSuccess()).
+      error(this.transactionFailed());
+  },
+
+  transactionSuccess () {
+    return (data, status) => {
+      this.redirectTo(this.followUpUrl);
+    }
+  },
+
+  transactionFailed () {
+    return (data, status) => {
+      this.enableButton();
+      let $errors = this.$('.fundraiser-bar__errors');
+      $errors.removeClass('hidden-closed');
+      $errors.find('.fundraiser-bar__error-detail').remove();
+      if (data.status == 422 && data.responseJSON && data.responseJSON.errors) {
+        var messages = data.responseJSON.errors.map(function(error){
```

Comment:
`let` is block scoped, `var` is function scoped. I want to use messages outside of the conditional block, so I'm using `var`.